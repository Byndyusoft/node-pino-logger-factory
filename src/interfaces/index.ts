export * from "./pino-http-logger-factory.interface";
export * from "./pino-http-logger-options-builder.interface";
export * from "./pino-logger-factory.interface";
export * from "./pino-logger-options-builder.interface";
