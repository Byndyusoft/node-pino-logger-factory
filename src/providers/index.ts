export * from "./pino-http-logger-factory";
export * from "./pino-http-logger-options-builder";
export * from "./pino-logger-factory";
export * from "./pino-logger-options-builder";
