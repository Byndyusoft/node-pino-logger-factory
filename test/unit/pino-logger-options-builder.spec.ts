import pino from "pino";
import {stdout} from "stdout-stderr";

import {LogLevel, PinoLoggerOptionsBuilder} from "../../src";

describe("pino-logger-options-builder", () => {
  afterEach(() => {
    delete process.env.CASC_ENV;
    delete process.env.BUILD_COMMIT;
  });

  it("must build options with defaults", () => {
    const builder = new PinoLoggerOptionsBuilder();

    expect(builder.build()).toMatchSnapshot();
  });

  it("must build options without defaults", () => {
    const builder = new PinoLoggerOptionsBuilder(false);

    expect(builder.build()).toMatchSnapshot();
  });

  it("must log custom error", () => {
    const builder = new PinoLoggerOptionsBuilder();

    stdout.start();
    pino(builder.build()).error({err: "some error"});
    stdout.stop();

    expect(JSON.parse(stdout.output)).toMatchSnapshot();
  });

  it("must log error", () => {
    const error = new Error("some error");
    error.stack = "Error: some error";

    const builder = new PinoLoggerOptionsBuilder();

    stdout.start();
    pino(builder.build()).error(error);
    stdout.stop();

    expect(JSON.parse(stdout.output)).toMatchSnapshot();
  });

  it("must log error without trace", () => {
    const error = new Error("some error");
    delete error.stack;

    const builder = new PinoLoggerOptionsBuilder();

    stdout.start();
    pino(builder.build()).error(error);
    stdout.stop();

    expect(JSON.parse(stdout.output)).toMatchSnapshot();
  });

  it("must log message", () => {
    const builder = new PinoLoggerOptionsBuilder();

    stdout.start();
    pino(builder.build()).info("hello");
    stdout.stop();

    expect(JSON.parse(stdout.output)).toMatchSnapshot();
  });

  it("must log message with args", () => {
    const builder = new PinoLoggerOptionsBuilder();

    stdout.start();
    pino(builder.build()).info("hello %s", "world");
    stdout.stop();

    expect(JSON.parse(stdout.output)).toMatchSnapshot();
  });

  it("must log object", () => {
    const builder = new PinoLoggerOptionsBuilder(false);

    stdout.start();
    pino(builder.build()).info({a: 1});
    stdout.stop();

    expect(JSON.parse(stdout.output)).toMatchSnapshot();
  });

  it("must log object and message", () => {
    const builder = new PinoLoggerOptionsBuilder();

    stdout.start();
    pino(builder.build()).info({a: 1}, "hello");
    stdout.stop();

    expect(JSON.parse(stdout.output)).toMatchSnapshot();
  });

  it("must log object and message with args", () => {
    const builder = new PinoLoggerOptionsBuilder();

    stdout.start();
    pino(builder.build()).info({a: 1}, "hello %s", "world");
    stdout.stop();

    expect(JSON.parse(stdout.output)).toMatchSnapshot();
  });

  it("must override level", () => {
    const builder = new PinoLoggerOptionsBuilder(false);

    expect(builder.withLevel(LogLevel.warn).build()).toMatchSnapshot();
  });

  it("must override pretty print", () => {
    const builder = new PinoLoggerOptionsBuilder(false);

    expect(builder.withPrettyPrint(true).build()).toMatchSnapshot();
  });

  it("must use process.env that starts with BUILD_", () => {
    process.env.BUILD_COMMIT = "eefc20";

    const builder = new PinoLoggerOptionsBuilder();

    expect(builder.build()).toMatchSnapshot();
  });

  it("must use process.env.CASC_ENV if exists", () => {
    process.env.CASC_ENV = "dev";

    const builder = new PinoLoggerOptionsBuilder();

    expect(builder.build()).toMatchSnapshot();
  });
});
