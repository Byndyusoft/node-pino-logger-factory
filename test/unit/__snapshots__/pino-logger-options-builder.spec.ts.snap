// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pino-logger-options-builder must build options with defaults 1`] = `
Object {
  "base": Object {
    "env": "test",
    "hostname": "host",
    "name": "name",
    "pid": 123456,
    "version": "version",
  },
  "formatters": Object {
    "level": [Function],
  },
  "hooks": Object {
    "logMethod": [Function],
  },
  "level": "info",
  "prettyPrint": false,
  "redact": Array [
    "req.headers.authorization",
  ],
  "serializers": Object {
    "err": [Function],
    "req": [Function],
    "res": [Function],
  },
  "timestamp": [Function],
}
`;

exports[`pino-logger-options-builder must build options without defaults 1`] = `
Object {
  "base": Object {},
  "formatters": Object {
    "level": [Function],
  },
  "hooks": Object {
    "logMethod": [Function],
  },
  "level": "info",
  "prettyPrint": false,
  "redact": Array [],
  "serializers": Object {},
  "timestamp": [Function],
}
`;

exports[`pino-logger-options-builder must log custom error 1`] = `
Object {
  "env": "test",
  "err": "some error",
  "hostname": "host",
  "level": "error",
  "name": "name",
  "pid": 123456,
  "time": "2016-04-05T17:02:19.796Z",
  "version": "version",
}
`;

exports[`pino-logger-options-builder must log error 1`] = `
Object {
  "env": "test",
  "err": Object {
    "message": "some error",
    "stack": "Error: some error",
    "type": "Error",
  },
  "errHash": "932f64f7",
  "hostname": "host",
  "level": "error",
  "name": "name",
  "pid": 123456,
  "time": "2016-04-05T17:02:19.796Z",
  "version": "version",
}
`;

exports[`pino-logger-options-builder must log error without trace 1`] = `
Object {
  "env": "test",
  "err": Object {
    "message": "some error",
    "type": "Error",
  },
  "errHash": "0",
  "hostname": "host",
  "level": "error",
  "name": "name",
  "pid": 123456,
  "time": "2016-04-05T17:02:19.796Z",
  "version": "version",
}
`;

exports[`pino-logger-options-builder must log message 1`] = `
Object {
  "env": "test",
  "hostname": "host",
  "level": "info",
  "msg": "hello",
  "msgTemplateHash": "248bfa47",
  "name": "name",
  "pid": 123456,
  "time": "2016-04-05T17:02:19.796Z",
  "version": "version",
}
`;

exports[`pino-logger-options-builder must log message with args 1`] = `
Object {
  "env": "test",
  "hostname": "host",
  "level": "info",
  "msg": "hello world",
  "msgTemplateHash": "3504eb44",
  "name": "name",
  "pid": 123456,
  "time": "2016-04-05T17:02:19.796Z",
  "version": "version",
}
`;

exports[`pino-logger-options-builder must log object 1`] = `
Object {
  "a": 1,
  "level": "info",
  "time": "2016-04-05T17:02:19.796Z",
}
`;

exports[`pino-logger-options-builder must log object and message 1`] = `
Object {
  "a": 1,
  "env": "test",
  "hostname": "host",
  "level": "info",
  "msg": "hello",
  "msgTemplateHash": "248bfa47",
  "name": "name",
  "pid": 123456,
  "time": "2016-04-05T17:02:19.796Z",
  "version": "version",
}
`;

exports[`pino-logger-options-builder must log object and message with args 1`] = `
Object {
  "a": 1,
  "env": "test",
  "hostname": "host",
  "level": "info",
  "msg": "hello world",
  "msgTemplateHash": "3504eb44",
  "name": "name",
  "pid": 123456,
  "time": "2016-04-05T17:02:19.796Z",
  "version": "version",
}
`;

exports[`pino-logger-options-builder must override level 1`] = `
Object {
  "base": Object {},
  "formatters": Object {
    "level": [Function],
  },
  "hooks": Object {
    "logMethod": [Function],
  },
  "level": "warn",
  "prettyPrint": false,
  "redact": Array [],
  "serializers": Object {},
  "timestamp": [Function],
}
`;

exports[`pino-logger-options-builder must override pretty print 1`] = `
Object {
  "base": Object {},
  "formatters": Object {
    "level": [Function],
  },
  "hooks": Object {
    "logMethod": [Function],
  },
  "level": "info",
  "prettyPrint": Object {
    "colorize": true,
    "ignore": "hash",
    "suppressFlushSyncWarning": true,
    "translateTime": "SYS:STANDARD",
  },
  "redact": Array [],
  "serializers": Object {},
  "timestamp": [Function],
}
`;

exports[`pino-logger-options-builder must use process.env that starts with BUILD_ 1`] = `
Object {
  "base": Object {
    "commit": "eefc20",
    "env": "test",
    "hostname": "host",
    "name": "name",
    "pid": 123456,
    "version": "version",
  },
  "formatters": Object {
    "level": [Function],
  },
  "hooks": Object {
    "logMethod": [Function],
  },
  "level": "info",
  "prettyPrint": false,
  "redact": Array [
    "req.headers.authorization",
  ],
  "serializers": Object {
    "err": [Function],
    "req": [Function],
    "res": [Function],
  },
  "timestamp": [Function],
}
`;

exports[`pino-logger-options-builder must use process.env.CASC_ENV if exists 1`] = `
Object {
  "base": Object {
    "env": "dev",
    "hostname": "host",
    "name": "name",
    "pid": 123456,
    "version": "version",
  },
  "formatters": Object {
    "level": [Function],
  },
  "hooks": Object {
    "logMethod": [Function],
  },
  "level": "info",
  "prettyPrint": false,
  "redact": Array [
    "req.headers.authorization",
  ],
  "serializers": Object {
    "err": [Function],
    "req": [Function],
    "res": [Function],
  },
  "timestamp": [Function],
}
`;
